app/packages/spaces/src/SpaceTree.ts

```diff
@@ -26,6 +26,10 @@ export default class SpaceTree {
       ? spaceNodeFromJSON(serializedTree)
       : new SpaceNode("root");
     if (onTreeUpdate) this.onUpdate = onTreeUpdate;
+
+    if (!this.root.hasActiveChild() && this.root.hasChildren()) {
```

Comment:
Is moving this logic from `Space` to `SpaceTree` a must? If so, we may need to traverse children to set active child for each underlying spaces.

I believe, if the root node is a space container instead of pane container, it will not set first panel as the active panel. In addition, if there is multiple spaces, it will also not set the first child as the active child for each space